key.ui/src/main/java/de/uka/ilkd/key/gui/JmlEnabledKeysIndicator.java

```diff
@@ -0,0 +1,42 @@
+/* This file is part of KeY - https://key-project.org
+ * KeY is licensed under the GNU General Public License Version 2
+ * SPDX-License-Identifier: GPL-2.0-only */
+package de.uka.ilkd.key.gui;
+
+import java.util.List;
+import javax.swing.*;
+
+import de.uka.ilkd.key.gui.extension.api.KeYGuiExtension;
+import de.uka.ilkd.key.settings.GeneralSettings;
+import de.uka.ilkd.key.settings.ProofIndependentSettings;
+
+/**
+ * Provides a Label in the status line for the indication of the current enabled keys for the JML
+ * condition evaluation.
+ *
+ * @author weigl
+ * @see GeneralSettings#getJmlEnabledKeys()
+ */
+@KeYGuiExtension.Info(experimental = false, name = "JML Enabled Keys Indicator for the Status Line")
+public class JmlEnabledKeysIndicator implements KeYGuiExtension, KeYGuiExtension.StatusLine {
+    private final GeneralSettings settings =
+        ProofIndependentSettings.DEFAULT_INSTANCE.getGeneralSettings();
+    private final JLabel lblIndicator = new JLabel();
+
+    public JmlEnabledKeysIndicator() {
+        lblIndicator.setBorder(BorderFactory.createEmptyBorder(5, 5, 5, 5));
+        lblIndicator.setToolTipText("Currently enabled keys for the JML condition evaluation");
+        settings.addPropertyChangeListener(GeneralSettings.KEY_JML_ENABLED_KEYS, it -> update());
+        update();
+    }
+
+    private void update() {
+        var lbl = String.join(", ", settings.getJmlEnabledKeys());
+        lblIndicator.setText(lbl);
+    }
+
+    @Override
+    public List<JComponent> getStatusLineComponents() {
```

Comment:
I'd have another small request: Following the design principle of the "dark cockpit philosophy", one should only indicate those things which are out of the ordinary. In 99.9% of the cases, we will use the default setting here. So it should be displayed, only if the deviate from the standard.

That would probably be just one if statement here.